src/app/personside/dialogpanel/fortsettDialog/leggTilbakePanel/LeggTilbakepanel.tsx

```diff
@@ -92,39 +94,54 @@ function LeggTilbakepanel(props: Props) {
-        if (leggerTilbake) {
+        if (props.status.type === DialogPanelStatus.POSTING) {
             return;
         }
         const callback = () => {
             dispatch(resetPlukkOppgaveResource);
             dispatch(reloadTildelteOppgaver);
         };
         if (LeggTilbakeValidator.erGyldigInnhabilRequest(state)) {
-            dispatch(
-                leggTilbakeResource.actions.post({ oppgaveId: props.oppgave.oppgaveid, type: 'Innhabil' }, callback)
-            );
+            props.setDialogStatus({ type: DialogPanelStatus.POSTING });
+            const payload: LeggTilbakeOppgaveRequest = { oppgaveId: props.oppgave.oppgaveid, type: 'Innhabil' };
+            post(`${apiBaseUri}/oppgaver/legg-tilbake`, payload)
+                .then(() => {
+                    callback();
+                    props.setDialogStatus({ type: DialogPanelStatus.OPPGAVE_LAGT_TILBAKE, payload: payload });
+                })
+                .catch(() => {
+                    props.setDialogStatus({ type: DialogPanelStatus.ERROR });
+                });
         } else if (LeggTilbakeValidator.erGyldigAnnenAarsakRequest(state)) {
-            dispatch(
-                leggTilbakeResource.actions.post(
-                    {
-                        beskrivelse: state.tekst,
-                        oppgaveId: props.oppgave.oppgaveid,
-                        type: 'AnnenAarsak'
-                    },
-                    callback
-                )
-            );
+            props.setDialogStatus({ type: DialogPanelStatus.POSTING });
+            const payload: LeggTilbakeOppgaveRequest = {
+                beskrivelse: state.tekst,
+                oppgaveId: props.oppgave.oppgaveid,
+                type: 'AnnenAarsak'
+            };
+            post(`${apiBaseUri}/oppgaver/legg-tilbake`, payload)
+                .then(() => {
+                    callback();
+                    props.setDialogStatus({ type: DialogPanelStatus.OPPGAVE_LAGT_TILBAKE, payload: payload });
+                })
+                .catch(() => {
+                    props.setDialogStatus({ type: DialogPanelStatus.ERROR });
+                });
         } else if (LeggTilbakeValidator.erGyldigFeilTemaRequest(state) && state.tema) {
-            dispatch(
-                leggTilbakeResource.actions.post(
-                    {
-                        temagruppe: state.tema.kodeRef,
-                        oppgaveId: props.oppgave.oppgaveid,
-                        type: 'FeilTema'
-                    },
-                    callback
-                )
-            );
+            props.setDialogStatus({ type: DialogPanelStatus.POSTING });
+            const payload: LeggTilbakeOppgaveRequest = {
+                temagruppe: state.tema.kodeRef,
+                oppgaveId: props.oppgave.oppgaveid,
+                type: 'FeilTema'
+            };
+            post(`${apiBaseUri}/oppgaver/legg-tilbake`, payload)
+                .then(() => {
```

Comment:
ble mye gjenntakende logikk her, trekke ut denne i en funksjon..?

Reply:
Trekke ut hele post i en egen funksjon? Det er så mange useStates og andre const som er denne er avhengig at det føler det blir litt overkill. Hver post-request har noe gjenntakende logikk men også mye forskjellig som kan føre til en litt for stor funksjon? Kan ta en prat om det

Reply:
ok, vi kan se på det :-)